src/main/handlers/saveProjectHandler.ts

```diff
@@ -32,15 +40,35 @@ const saveProjectToFile: (
   await writeFile(filePath, projectAsString);
 };
 
-const handleSaveProject: (
+export const handleSaveProject: (
   mainWindow: BrowserWindow | null,
   project: Project
 ) => Promise<string> = async (mainWindow, project) => {
-  const filePath = await getSaveFilePath(mainWindow);
+  const filePath =
```

Comment:
Similar thingo with the `??` operator in place of a null-comparing ternary
```suggestion
  const filePath = project.projectFilePath ?? await getSaveFilePath(mainWindow, project.name)
```